elastic/apm-server
# testing/benchmark: Allow setting apm index shards

## Motivation/summary

Introduces a new variable `apm_shards` which allows tunning the APM
component templates `number_of_shards` setting, with no default.

This allows us to benchmark the APM Server with a more suitable config
as the APM Server size increases, and also test different permutations
of config options.

Also defaults the benchmarking to always use the `integrations_server`
resource and tunes the 16GB APM Server profile.

## Checklist

~- [ ] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)~
~- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)~

## How to test these changes

Set the shard setting to something different than 1 and check that the component templates are updated.

## Related issues

Part of #8775
